### ISSUE TITLE
2021 week 32: rebase off upstream

### ISSUE THREAD

**Issue by adpaco-aws** [model-checking/rmc#380]

**Comment by adpaco-aws** [comment]
No conflicts during rebase.

**Comment by adpaco-aws** [comment]
We get the same error in several tests:
```
running 198 tests
.iii.........................i.i..................F................................................. 100/198
......i..........ii..i......i...ii..F......i....i....i.i.....i........F.......i.iiiF.F......i..F..
failures:

---- [rmc] cbmc/CopyIntrinsics/main.rs stdout ----

error: test failed: expected codegen success, got failure
status: exit status: 101
command: "rmc-rustc" "-Z" "codegen-backend=gotoc" "--cfg=rmc" "--out-dir" "/home/runner/work/rmc/rmc/build/x86_64-unknown-linux-gnu/test/cbmc/CopyIntrinsics/main" "/home/runner/work/rmc/rmc/src/test/cbmc/CopyIntrinsics/main.rs"
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
error: internal compiler error: trimmed_def_paths constructed
  |
  = note: delayed at    0: rustc_errors::HandlerInner::delay_good_path_bug
                       at ./compiler/rustc_errors/src/lib.rs:1088:25
             1: rustc_errors::Handler::delay_good_path_bug
                       at ./compiler/rustc_errors/src/lib.rs:784:9
             2: rustc_middle::ty::print::pretty::trimmed_def_paths
                       at ./compiler/rustc_middle/src/ty/print/pretty.rs:2380:9
             3: rustc_query_system::query::plumbing::try_execute_query::{{closure}}
                       at ./compiler/rustc_query_system/src/query/plumbing.rs:461:55
             4: stacker::maybe_grow
                       at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.12/src/lib.rs:55:9
             5: rustc_data_structures::stack::ensure_sufficient_stack
                       at ./compiler/rustc_data_structures/src/stack.rs:16:5
             6: <rustc_query_impl::plumbing::QueryCtxt as rustc_query_system::query::QueryContext>::start_query::{{closure}}::{{closure}}
                       at ./compiler/rustc_query_impl/src/plumbing.rs:137:17
             7: rustc_middle::ty::context::tls::enter_context::{{closure}}
                       at ./compiler/rustc_middle/src/ty/context.rs:1782:50
             8: rustc_middle::ty::context::tls::set_tlv
                       at ./compiler/rustc_middle/src/ty/context.rs:1766:9
             9: rustc_middle::ty::context::tls::enter_context
                       at ./compiler/rustc_middle/src/ty/context.rs:1782:9
            10: <rustc_query_impl::plumbing::QueryCtxt as rustc_query_system::query::QueryContext>::start_query::{{closure}}
                       at ./compiler/rustc_query_impl/src/plumbing.rs:136:13
            11: rustc_middle::ty::context::tls::with_related_context::{{closure}}
                       at ./compiler/rustc_middle/src/ty/context.rs:1826:13
            12: rustc_middle::ty::context::tls::with_context::{{closure}}
                       at ./compiler/rustc_middle/src/ty/context.rs:1810:40
            13: rustc_middle::ty::context::tls::with_context_opt
                       at ./compiler/rustc_middle/src/ty/context.rs:1799:22
            14: rustc_middle::ty::context::tls::with_context
                       at ./compiler/rustc_middle/src/ty/context.rs:1810:9
            15: rustc_middle::ty::context::tls::with_related_context
                       at ./compiler/rustc_middle/src/ty/context.rs:1823:9
            16: <rustc_query_impl::plumbing::QueryCtxt as rustc_query_system::query::QueryContext>::start_query
                       at ./compiler/rustc_query_impl/src/plumbing.rs:125:9
            17: rustc_query_system::query::plumbing::try_execute_query
                       at ./compiler/rustc_query_system/src/query/plumbing.rs:461:22
            18: rustc_query_system::query::plumbing::get_query_impl
                       at ./compiler/rustc_query_system/src/query/plumbing.rs:709:5
            19: rustc_query_system::query::plumbing::get_query
                       at ./compiler/rustc_query_system/src/query/plumbing.rs:832:17
            20: rustc_middle::ty::query::TyCtxtAt::trimmed_def_paths
                       at ./compiler/rustc_middle/src/ty/query.rs:203:17
            21: rustc_middle::ty::query::<impl rustc_middle::ty::context::TyCtxt>::trimmed_def_paths
                       at ./compiler/rustc_middle/src/ty/query.rs:184:17
            22: rustc_middle::ty::print::pretty::PrettyPrinter::try_print_trimmed_def_path
                       at ./compiler/rustc_middle/src/ty/print/pretty.rs:288:15
            23: <rustc_middle::ty::print::pretty::FmtPrinter<F> as rustc_middle::ty::print::Printer>::print_def_path
                       at ./compiler/rustc_middle/src/ty/print/pretty.rs:1385:19
            24: <rustc_middle::ty::print::pretty::FmtPrinter<F> as rustc_middle::ty::print::pretty::PrettyPrinter>::print_value_path
                       at ./compiler/rustc_middle/src/ty/print/pretty.rs:1586:16
            25: rustc_middle::ty::print::pretty::PrettyPrinter::pretty_print_const
                       at ./compiler/rustc_middle/src/ty/print/pretty.rs:937:29
            26: rustc_middle::mir::pretty_print_const::{{closure}}
                       at ./compiler/rustc_middle/src/mir/mod.rs:2773:9
            27: rustc_middle::ty::context::tls::with::{{closure}}
                       at ./compiler/rustc_middle/src/ty/context.rs:1837:32
            28: rustc_middle::ty::context::tls::with_context::{{closure}}
                       at ./compiler/rustc_middle/src/ty/context.rs:1810:40
            29: rustc_middle::ty::context::tls::with_context_opt
                       at ./compiler/rustc_middle/src/ty/context.rs:1799:22
            30: rustc_middle::ty::context::tls::with_context
                       at ./compiler/rustc_middle/src/ty/context.rs:1810:9
            31: rustc_middle::ty::context::tls::with
                       at ./compiler/rustc_middle/src/ty/context.rs:1837:9
            32: rustc_middle::mir::pretty_print_const
                       at ./compiler/rustc_middle/src/mir/mod.rs:2769:5
            33: core::fmt::write
                       at ./library/core/src/fmt/mod.rs:1117:17
            34: core::fmt::Formatter::write_fmt
                       at ./library/core/src/fmt/mod.rs:1563:9
            35: <rustc_middle::mir::Constant as core::fmt::Debug>::fmt
                       at ./compiler/rustc_middle/src/mir/mod.rs:2740:9
            36: core::fmt::write
                       at ./library/core/src/fmt/mod.rs:1117:17
            37: core::fmt::Formatter::write_fmt
                       at ./library/core/src/fmt/mod.rs:1563:9
            38: <rustc_middle::mir::Operand as core::fmt::Debug>::fmt
            39: core::fmt::run
                       at ./library/core/src/fmt/mod.rs:1162:5
            40: core::fmt::write
                       at ./library/core/src/fmt/mod.rs:1130:26
            41: core::fmt::Formatter::write_fmt
                       at ./library/core/src/fmt/mod.rs:1563:9
            42: <rustc_middle::mir::AssertKind<O> as core::fmt::Debug>::fmt
            43: <&T as core::fmt::Debug>::fmt
                       at ./library/core/src/fmt/mod.rs:2038:62
            44: core::fmt::write
                       at ./library/core/src/fmt/mod.rs:1117:17
            45: core::fmt::Write::write_fmt
                       at ./library/core/src/fmt/mod.rs:187:9
            46: alloc::fmt::format
                       at ./library/alloc/src/fmt.rs:583:5
            47: rustc_codegen_llvm::gotoc::mir_to_goto::codegen::statement::<impl rustc_codegen_llvm::gotoc::mir_to_goto::context::goto_ctx::GotocCtx>::codegen_terminator
                       at ./compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/statement.rs:78:45
            48: rustc_codegen_llvm::gotoc::mir_to_goto::codegen::block::<impl rustc_codegen_llvm::gotoc::mir_to_goto::context::goto_ctx::GotocCtx>::codegen_block
                       at ./compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/block.rs:30:28
            49: rustc_codegen_llvm::gotoc::mir_to_goto::codegen::function::<impl rustc_codegen_llvm::gotoc::mir_to_goto::context::goto_ctx::GotocCtx>::codegen_function::{{closure}}
                       at ./compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/function.rs:95:71
            50: core::iter::traits::iterator::Iterator::for_each::call::{{closure}}
                       at ./library/core/src/iter/traits/iterator.rs:730:29
            51: core::iter::adapters::map::map_fold::{{closure}}
                       at ./library/core/src/iter/adapters/map.rs:84:21
            52: <core::iter::adapters::enumerate::Enumerate<I> as core::iter::traits::iterator::Iterator>::fold::enumerate::{{closure}}
                       at ./library/core/src/iter/adapters/enumerate.rs:106:27
            53: core::iter::traits::iterator::Iterator::fold
                       at ./library/core/src/iter/traits/iterator.rs:2170:21
            54: <core::iter::adapters::enumerate::Enumerate<I> as core::iter::traits::iterator::Iterator>::fold
                       at ./library/core/src/iter/adapters/enumerate.rs:112:9
            55: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
                       at ./library/core/src/iter/adapters/map.rs:124:9
            56: core::iter::traits::iterator::Iterator::for_each
                       at ./library/core/src/iter/traits/iterator.rs:733:9
            57: rustc_codegen_llvm::gotoc::mir_to_goto::codegen::function::<impl rustc_codegen_llvm::gotoc::mir_to_goto::context::goto_ctx::GotocCtx>::codegen_function
                       at ./compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/function.rs:95:13
            58: <rustc_codegen_llvm::gotoc::mir_to_goto::compiler_interface::GotocCodegenBackend as rustc_codegen_ssa::traits::backend::CodegenBackend>::codegen_crate::{{closure}}
                       at ./compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/compiler_interface.rs:97:35
            59: rustc_codegen_llvm::gotoc::mir_to_goto::utils::debug::<impl rustc_codegen_llvm::gotoc::mir_to_goto::context::goto_ctx::GotocCtx>::call_with_panic_debug_info::{{closure}}
                       at ./compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/utils/debug.rs:77:13
            60: std::thread::local::LocalKey<T>::try_with
                       at ./library/std/src/thread/local.rs:399:16
            61: std::thread::local::LocalKey<T>::with
                       at ./library/std/src/thread/local.rs:375:9
            62: rustc_codegen_llvm::gotoc::mir_to_goto::utils::debug::<impl rustc_codegen_llvm::gotoc::mir_to_goto::context::goto_ctx::GotocCtx>::call_with_panic_debug_info
                       at ./compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/utils/debug.rs:75:9
            63: <rustc_codegen_llvm::gotoc::mir_to_goto::compiler_interface::GotocCodegenBackend as rustc_codegen_ssa::traits::backend::CodegenBackend>::codegen_crate
                       at ./compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/compiler_interface.rs:96:25
            64: rustc_interface::passes::start_codegen::{{closure}}
                       at ./compiler/rustc_interface/src/passes.rs:1053:9
            65: rustc_data_structures::profiling::VerboseTimingGuard::run
                       at ./compiler/rustc_data_structures/src/profiling.rs:611:9
            66: rustc_session::utils::<impl rustc_session::session::Session>::time
                       at ./compiler/rustc_session/src/utils.rs:16:9
            67: rustc_interface::passes::start_codegen
                       at ./compiler/rustc_interface/src/passes.rs:1052:19
            68: rustc_interface::queries::Queries::ongoing_codegen::{{closure}}::{{closure}}
                       at ./compiler/rustc_interface/src/queries.rs:254:20
            69: rustc_interface::passes::QueryContext::enter::{{closure}}
                       at ./compiler/rustc_interface/src/passes.rs:779:42
            70: rustc_middle::ty::context::tls::enter_context::{{closure}}
                       at ./compiler/rustc_middle/src/ty/context.rs:1782:50
            71: rustc_middle::ty::context::tls::set_tlv
                       at ./compiler/rustc_middle/src/ty/context.rs:1766:9
            72: rustc_middle::ty::context::tls::enter_context
                       at ./compiler/rustc_middle/src/ty/context.rs:1782:9
            73: rustc_interface::passes::QueryContext::enter
                       at ./compiler/rustc_interface/src/passes.rs:779:9
            74: rustc_interface::queries::Queries::ongoing_codegen::{{closure}}
                       at ./compiler/rustc_interface/src/queries.rs:245:13
            75: rustc_interface::queries::Query<T>::compute
                       at ./compiler/rustc_interface/src/queries.rs:38:28
            76: rustc_interface::queries::Queries::ongoing_codegen
                       at ./compiler/rustc_interface/src/queries.rs:243:9
            77: rustc_driver::run_compiler::{{closure}}::{{closure}}
                       at ./compiler/rustc_driver/src/lib.rs:407:13
            78: rustc_interface::queries::<impl rustc_interface::interface::Compiler>::enter
                       at ./compiler/rustc_interface/src/queries.rs:394:19
            79: rustc_driver::run_compiler::{{closure}}
                       at ./compiler/rustc_driver/src/lib.rs:312:22
            80: rustc_interface::interface::create_compiler_and_run::{{closure}}
                       at ./compiler/rustc_interface/src/interface.rs:209:13
            81: rustc_span::with_source_map
                       at ./compiler/rustc_span/src/lib.rs:911:5
            82: rustc_interface::interface::create_compiler_and_run
                       at ./compiler/rustc_interface/src/interface.rs:203:5
            83: rustc_interface::interface::run_compiler::{{closure}}
                       at ./compiler/rustc_interface/src/interface.rs:225:12
            84: rustc_interface::util::setup_callbacks_and_run_in_thread_pool_with_globals::{{closure}}::{{closure}}
                       at ./compiler/rustc_interface/src/util.rs:157:13
            85: scoped_tls::ScopedKey<T>::set
                       at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137:9
            86: rustc_span::create_session_globals_then
                       at ./compiler/rustc_span/src/lib.rs:105:5
            87: rustc_interface::util::setup_callbacks_and_run_in_thread_pool_with_globals::{{closure}}
                       at ./compiler/rustc_interface/src/util.rs:155:9
            88: rustc_interface::util::scoped_thread::{{closure}}
                       at ./compiler/rustc_interface/src/util.rs:130:24
            89: std::sys_common::backtrace::__rust_begin_short_backtrace
                       at ./library/std/src/sys_common/backtrace.rs:125:18
            90: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
                       at ./library/std/src/thread/mod.rs:481:17
            91: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
                       at ./library/core/src/panic/unwind_safe.rs:271:9
            92: std::panicking::try::do_call
                       at ./library/std/src/panicking.rs:403:40
            93: std::panicking::try
                       at ./library/std/src/panicking.rs:367:19
            94: std::panic::catch_unwind
                       at ./library/std/src/panic.rs:129:14
            95: std::thread::Builder::spawn_unchecked::{{closure}}
                       at ./library/std/src/thread/mod.rs:480:30
            96: core::ops::function::FnOnce::call_once{{vtable.shim}}
                       at ./library/core/src/ops/function.rs:227:5
            97: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
                       at ./library/alloc/src/boxed.rs:1636:9
            98: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
                       at ./library/alloc/src/boxed.rs:1636:9
            99: std::sys::unix::thread::Thread::new::thread_start
                       at ./library/std/src/sys/unix/thread.rs:106:17
           100: start_thread
           101: clone
          

thread 'rustc' panicked at 'no warnings or errors encountered even though `delayed_good_path_bugs` issued', compiler/rustc_errors/src/lib.rs:1134:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.56.0-dev running on x86_64-unknown-linux-gnu

note: compiler flags: -Z codegen-backend=gotoc

query stack during panic:
end of query stack

[RMC] no current codegen item.

RMC unexpectedly panicked during code generation.
```

**Comment by adpaco-aws** [comment]
Pushed failing branch [main-154-2021-08-17](https://github.com/model-checking/rmc/tree/main-154-2021-08-17). Opened #433 . Fix for it will follow soon.

**Comment by adpaco-aws** [comment]
Opened #432 for the fix and #434 to figure out why the flag is required.

**Comment by adpaco-aws** [comment]
[main-154-2021-08-17](https://github.com/model-checking/rmc/tree/main-154-2021-08-17) is now the default branch with #432 on top of it.
